lib/mlxdevm.c

```diff
@@ -183,6 +183,67 @@ mlxdevm_sf_port_add(struct mlxdevm *dl, uint32_t pfnum, uint32_t sfnum)
 	return NULL;
 }
 
+static int cmd_port_dump_cb_to_list(const struct nlmsghdr *nlh, void *data)
+{
+	struct genlmsghdr *genl = mnl_nlmsg_get_payload(nlh);
+	struct nlattr *tb[MLXDEVM_ATTR_MAX + 1] = {};
+	struct port_list_head *head = data;
+	struct mlxdevm_port_list *cur;
+	struct mlxdevm_port *port;
+
+	mnl_attr_parse(nlh, sizeof(*genl), attr_cb, tb);
+	if (!tb[MLXDEVM_ATTR_DEV_BUS_NAME] || !tb[MLXDEVM_ATTR_DEV_NAME] ||
+	    !tb[MLXDEVM_ATTR_PORT_INDEX] || !tb[MLXDEVM_ATTR_PORT_PCI_PF_NUMBER] ||
```

Comment:
mlxdevm returns all type of ports = physical, vf, pf, virtual etc.
so return error on 198 will not work.

so the code you need is,

flavour = mln_attr_get(tb[MLXDEVM_ATTR_PORT_FLAVOUR);
if (flavour != PCISF)
    return MNL_CB_OK;

/* This is for sure an SF port */
logic to add to the list.

Reply:
is PCISF "MLXDEVM_ATTR_PORT_PCI_SF_NUMBER"?